src/index.js

```diff
@@ -1,32 +1,43 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const API = 'https://rickandmortyapi.com/api/character/';
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
-    }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
+
+function fetchData(url_api) {
+  return new Promise( (resolv, reject)=>{
```

Comment:
- Tratemos de no dejar espacios que no son necesarios. 
- Tener cuidado con los nombres asignados: "### resolve"

```suggestion
  return new Promise((resolve, reject)=>{
```